physical/cockroachdb/cockroachdb.go

```diff
@@ -99,26 +132,45 @@ func NewCockroachDBBackend(conf map[string]string, logger log.Logger) (physical.
 			"list":   "SELECT path FROM " + dbTable + " WHERE path LIKE $1",
 		},
 		statements: make(map[string]*sql.Stmt),
-		logger:     logger,
-		permitPool: physical.NewPermitPool(maxParInt),
+		rawHAStatements: map[string]string{
+			"get": "SELECT ha_value FROM " + dbHATable + " WHERE NOW() <= valid_until AND ha_key = $1",
+			"upsert": "INSERT INTO " + dbHATable + " as t (ha_identity, ha_key, ha_value, valid_until)" +
+				" VALUES ($1, $2, $3, NOW() + $4) " +
+				" ON CONFLICT (ha_key) DO " +
+				" UPDATE SET (ha_identity, ha_key, ha_value, valid_until) = ($1, $2, $3, NOW() + $4) " +
```

Comment:
I know little about CockroachDB, but I thought it was fully compatible with PG.  You've been diligent about adapting the PG approach here, but I noticed this one discrepancy: instead of `NOW() + $4 * INTERVAL '1 seconds'` you have simply `NOW() + $4`.  Why's that?

Reply:
What's happening here is that `$4` is being passed in as a string type, not an integer type. When we attempt to to `$4 * INTERVAL '1 seconds'`, we get a conversion error:

```
ERROR: unsupported binary operator: <string> * <interval> (desired <interval>)
SQLSTATE: 22023
```

We would have the option of casting `$4` to an integer, and then multiplying by an interval: `NOW() + $4::int * INTERVAL '1 seconds'`.

However, $4 is already a value in seconds, and so casting to an int just to multiply by 1 second is unnecessary. It's more succinct to just add the string value directly to `NOW()`.